lib/Fhp/Action/SendSEPADirectDebit.php

```diff
@@ -57,6 +70,12 @@ public static function create(SEPAAccount $account, string $painMessage): SendSE
             $ctrlSum = $matches['ctrlsum'];
         }
 
+		if (preg_match('/<PmtTpInf>.*<LclInstrm>.*<Cd>(?<coretype>CORE|COR1|B2B)<\/Cd>.*<\/LclInstrm>.*<\/PmtTpInf>/s', $painMessage, $matches) === 1) {
```

Comment:
If instead of `/` you pick a different delimiter character, e.g. `@`, you don't have to escape the slashes within the regex.